### PR TITLE
Congrats: disable toast notification

### PR DESCRIPTION
**Pull request by chad1008**
Related to 2736-gh-Automattic/dotcom-forge

## Proposed Changes

Disable toast notification for congrats pages. 

## Testing Instructions

Perform test purchases of plans, domains, and add-ons, confirming that none of them display a toast notification upon a successful purchase.

### CODE REVIEW

File: client/my-sites/checkout/checkout-thank-you/pending/index.tsx
Comment (stephanethomas) [inline client/my-sites/checkout/checkout-thank-you/pending/index.tsx]: Here is another approach for your consideration:

```javascript
if ( isRefactoredForThankYouV2 ) {
  // Disables notice as it is redundant with the content of the new Thank You page
  return;
}

reduxDispatch(
  successNotice( translate( 'Your purchase has been completed!' ), {
    id: 'checkout-thank-you-success',
    displayOnNextPage: true,
  } )
);
```

**Review by stephanethomas — APPROVED** [review]
Works great! I didn't see any notice when purchasing an additional email:

image

File: client/my-sites/checkout/checkout-thank-you/pending/index.tsx
Comment (wongasy) [reply client/my-sites/checkout/checkout-thank-you/pending/index.tsx]: I would probably get rid of the `isRefactoredForThankYouV2` check altogether. We should be good to remove this notice for both legacy and redesign Congrats Page. And in reality, I think the majority (if not all) of purchases land on the redesigned page. As discussed in Slack, I'm not even sure if the "delayed transfer" flow is valid at this point.

* When users purchase a paid plan + domain transfer, they are taken to the "What is your goal?" flow
* Users cannot purchase domain transfer when they select a free plan.

See p1712689984399789/1712687032.878589-slack-C06ELHR6L9J

Comment (chad1008) [reply client/my-sites/checkout/checkout-thank-you/pending/index.tsx]: I've added a new change that just removes the notification all together. I'll wait for CI and a final approval from you @wongasy!

**Review by wongasy — APPROVED** [review]
LGTM and tests well! 

I'm no longer seeing the "Your purchase has been completed!" notice after making a purchase. 

https://github.com/Automattic/wp-calypso/assets/104910361/936064a2-95c7-4e10-bd08-58841e50aad9

Noting that the Testing Instructions might be off. I tested purchasing a domain transfer + paid plan, and I was taken to an upsell page, then the "What is your goal" page. The "Purchase completed" notice was not shown. Given that that flow doesn't take users to the Congrats Page, the changes we're making in `<CheckoutThankYou>` is not relevant to that flow. To avoid confusion, it might make sense to remove the mentions of that scenario in the PR.

**Comment by chad1008** [comment]
Thanks again, @wongasy!

**Pull request merged**
